plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileOperations.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.collect.Multiset;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.filesystem.tracing.FileSystemAttributes.FILE_LOCATION;
+import static io.trino.plugin.hive.HiveQueryRunner.HIVE_CATALOG;
+import static io.trino.testing.MultisetAssertions.assertMultisetsEqual;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toCollection;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestHiveFileOperations
```

Comment:
Just an idea:
how about set directory lister to `CountingAccessDirectoryLister` explicitly to HiveQueryRunner, and use method invocation stats during flush procedure:

smth like this:

```
public class CountingAccessDirectoryLister
        implements DirectoryLister
{
    public enum Method
    {
        LIST_FILES,
        INVALIDATE_TABLE,
        INVALIDATE_PARTITION,
        INVALIDATE_ALL,
    }

    private final Multiset<Method> methodInvocations = ConcurrentHashMultiset.create();
    private final DirectoryLister delegate;

    public CountingAccessDirectoryLister(DirectoryLister delegate)
    {
        this.delegate = requireNonNull(delegate, "delagate is null");
    }

    public Multiset<Method> getMethodInvocations()
    {
        return ImmutableMultiset.copyOf(methodInvocations);
    }

    public void resetCounters()
    {
        methodInvocations.clear();
    }

    @Override
    public RemoteIterator<TrinoFileStatus> listFilesRecursively(TrinoFileSystem fs, Table table, Location location)
            throws IOException
    {
        methodInvocations.add(Method.LIST_FILES);
        return delegate.listFilesRecursively(fs, table, location);
    }

....
```

Reply:
I prefer to use the existing tracing mechanisms for tracking filesystem operations